test/iterable_basics_test.dart

```diff
@@ -349,6 +349,24 @@ void main() {
       expect(emptyInts.maxBy((a) => a.toString().length), isNull);
       expect(emptyStrings.maxBy((a) => a.length), isNull);
     });
+
+    test('does not calculate any sort key more than once', () {
+      final values = [3, 222, 11, 15, 18];
+
+      final callCounts = <int, int>{};
+      int recordCallAndReturn(int e) {
+        callCounts.putIfAbsent(e, () => 0);
+        callCounts[e] = callCounts[e]! + 1;
```

Comment:
totally optional

I think this could be one line:

```dart
callCounts.update(e, (value) => value++, ifAbsent: () => 1);
```

Reply:
Done